File: bench/iteration-recursion/2.go
package main

import (
	"os"
	"strconv"
	"fmt"
	"time"
)

// Recursion function
func rec(n uint64) uint64 {
	if n <= 0 {
		return n
	}

	return rec(n - 1)
}

func main() {
	var n uint64 = 1000
	var args = os.Args[1:]
	if len(args) >= 1 {
		n, _ = strconv.ParseUint(args[0], 10, 64)
	}

	var start = time.Now()

	n = rec(n)

	println(n)

	fmt.Println(time.Now().Sub(start).Nanoseconds())
}

File: bench/iteration-recursion/2.cpp
#include <iostream>
#include <chrono>

using namespace std;

// Recursion function
int rec(unsigned long long int n)
{
	if (n <= 0)
	{
		return n;
	}

	return rec(n - 1);
}

int main(int argc, char const *argv[])
{
	unsigned long long int n = 1000;
	if (argc == 2)
	{
		n = atoi(argv[1]);
	}

	auto start = chrono::steady_clock::now();

	n = rec(n);

	cout << n << endl;

	cout << chrono::duration_cast<chrono::nanoseconds>(chrono::steady_clock::now() - start).count() << endl;

	return 0;
}
